Objective-C/CBLQuery.mm

```diff
@@ -200,18 +209,24 @@ - (CBLQueryParameters*) parameters {
 
 - (void) setParameters: (CBLQueryParameters*)parameters {
     CBL_LOCK(self) {
-        if (parameters)
+        if (parameters) {
+            NSError* error = nil;
+            NSData* params = [parameters encode: &error];
+            if (!params)
+                [NSException raise: NSInvalidArgumentException
```

Comment:
Can you also log the error for the case that somebody experiences a crash due to this so they have some info in the log?